### PR TITLE
added README.md and data to InjectDataController

### PR DESCRIPTION
**Pull request by matviichuk92**

### CODE REVIEW

**Comment by matviichuk92** [comment]
[Ось як це буде виглядати](https://github.com/matviichuk92/Online-Shop/blob/2fa75dcccb04951199bcebd5500eaa08c2fa05c4/README.md)

File: README.md [hunk 1]
```diff
@@ -1,2 +1,38 @@
...
+**Project purpose**
+__________________
+Create online project on base localhost with connection to DataBase 
+with next functionality:
+- register users and save password in hashing view in DB;
+- access to pages according to roles(user and admin);
+- users can: look through products and add it to own cart, edit cart, 
+create orders and see all own orders;
+- admin have access to all information about users, orders, products with ability to edit and delete them.
+___________________________
```
Comment (AnaLanda) [inline README.md]:
```suggestion
An online project in the form of an internet shop prototype with a connection to a database (DB).
The shop presented in this project has the following functionality:
- registering users and saving hashed passwords in the DB;
- role-based access to pages for users and admins;
- users can: look through the products in the store and add them to their cart, edit the cart, 
place orders, and see all their previous orders;
- admins have access to all information about the users, their orders, and products in the store, which they can edit and delete.
___________________________
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,38 @@
...
 
-"# Online-Shop" 
+**Project structure and details**
+_________________
+Project have N-tier architecture:
```
Comment (AnaLanda) [inline README.md]:
```suggestion
The project has an N-tier architecture: 
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,38 @@
...
 
-"# Online-Shop" 
+**Project structure and details**
+_________________
+Project have N-tier architecture: 
+- controller(request processing and response sending);
+- service(business logic);
+- DAO(CRUD operation with DB);
+
+Also, four major entities: Product, User, ShoppingCart and Order.
```
Comment (AnaLanda) [inline README.md]:
```suggestion
The project also features four major entities: Product, User, ShoppingCart, and Order.
```

File: README.md [hunk 4]
```diff
@@ -1,2 +1,38 @@
...
+ 
+Project correspond to SOLID, KISS and DRY principles. It's created with Java EE, MySQL, Maven, Apache Tomcat.
```
Comment (AnaLanda) [inline README.md]:
```suggestion
The project follows SOLID, KISS, and DRY principles. It was created with Java EE, MySQL, Maven, and Apache Tomcat.
```

File: README.md [hunk 5]
```diff
@@ -1,2 +1,38 @@
...
+_______________ 
+**Launch guide**
+__________________
+- Fork project;
```
Comment (AnaLanda) [inline README.md]:
```suggestion
- Download the project to your machine;
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,38 @@
...
+- Add Maven;
+- Download Tomcat from _tomcat.apache.org_, unzip, add it to _install package_, add the new configuration
+(if you use IntelJ) TomCat Server(choose "Local" and point install package) in project, 
+choose the artifact _war-exploded_, in _application context_ field just put slash symbol;
+- Using MySQL Workbench 8.0 CE create a schema and tables as specified in init-db.sql;
+- Register like regular user;
```
Comment (AnaLanda) [inline README.md]:
```suggestion
- Launch the project using the Tomcat Server;
- Register as a user on the website;
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,38 @@
...
+- Log in to site and push button "Inject data" and check user functionality;
+- Logout and log in like admin(login - boss, password - 123) and check admin functionality;
```
Comment (AnaLanda) [inline README.md]:
```suggestion
- Log in to the website, click on the button "Inject data" and check the user functionality;
- Logout and log in as an admin using login "boss" and password "123" and check the admin functionality;
```

**Review by AnaLanda — COMMENTED** [review]
Looks good! I added a couple language-related suggestions)

**Comment by matviichuk92** [comment]
> Looks good! I added a couple language-related suggestions)

 I owe you one!

File: src/main/java/online/shop/controller/InjectDataController.java [hunk 8]
```diff
@@ -1,17 +1,37 @@
...
+            (UserService) injector.getInstance(UserService.class);
+    private final ProductService productService =
+            (ProductService) injector.getInstance(ProductService.class);
 
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
+        User admin = new User("Boss","boss","123");
+        userService.create(admin);
+        admin.setRoles(Set.of(Role.of("ADMIN")));
```
Comment (KseniiaMakarova) [inline src/main/java/online/shop/controller/InjectDataController.java]: you are saving admin and only then setting his roles

File: src/main/java/online/shop/controller/InjectDataController.java [hunk 9]
```diff
@@ -1,17 +1,37 @@
...
+        Product apple = new Product("apple",12.45);
```
Comment (KseniiaMakarova) [inline src/main/java/online/shop/controller/InjectDataController.java]:
```suggestion
        Product apple = new Product("apple", 12.45);
```
add whitespaces after comma everywhere needed

File: src/main/webapp/WEB-INF/views/inject-data.jsp [hunk 10]
```diff
@@ -4,7 +4,12 @@
     <title>Hello</title>
 </head>
 <body>
-<h1>You data will be adding to DB</h1>
+<center><h3>The data added to DB.</h3>
+    <h3>Admin login is boss .</h3>
+        <h3>Admin password is 123 .</h3>
+            <h3>Please, logout and enter as admin to see admin functionality </h3>
+</body>
+</center>
```
Comment (KseniiaMakarova) [inline src/main/webapp/WEB-INF/views/inject-data.jsp]:
```suggestion
</center>
</body>
```
what do you think?

File: README.md [hunk 11]
```diff
@@ -1,2 +1,39 @@
...
+- Download Tomcat from _tomcat.apache.org_, unzip, add it to _install package_, add the new configuration
+(if you use IntelJ) TomCat Server(choose "Local" and point install package) in project, 
+choose the artifact _war-exploded_, in _application context_ field just put slash symbol;
+- Using MySQL Workbench 8.0 CE create a schema and tables as specified in init-db.sql;
+- Launch the project using the Tomcat Server;
+- Log in to the website, click on the button "Inject data" and check the user functionality;
+- Logout and log in as an admin using login "boss" and password "123" and check the admin functionality;
+____________________________________
+**Roman Matviichuk**
+**matviihuk.office@gmail.com**
```
Comment (KseniiaMakarova) [inline README.md]: let's add a hyperlink to your GitHub profile here instead (see markdown cheatsheet)

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
also, you should remove all the branches except this one and `master`

**Pull request merged**